src/test/java/com/puppycrawl/tools/checkstyle/internal/TestUtils.java

```diff
@@ -56,4 +66,47 @@ public static PackageObjectFactory getPackageObjectFactory() throws CheckstyleEx
         final Set<String> packageNames = PackageNamesLoader.getPackageNames(cl);
         return new PackageObjectFactory(packageNames, cl);
     }
+
+    /**
+     * Finds node of specified type among root children, siblings, siblings children
+     * on any deep level.
```

Comment:
To me this method seems too specific. It searches current node and it's future siblings. It's main purpose is only to avoid previous siblings.
I can't imagine this being used very often.
@romani What is your opinion?